### ISSUE TITLE
Array.Clear for Generic.

### ISSUE THREAD

**Issue by iskiselev** [sq/JSIL#784]
Array.Clear fails for such test case:
```
using System;

public class GenericType<T>
{
    private InnerType[] innerArray = new InnerType[1] { new InnerType() };

    public void ClearInnerArray()
    {
        Array.Clear(innerArray, 0, 1);
    }

    private struct InnerType
    {
    }
}

public static class Program {
    public static void Main (string[] args) {
        new GenericType<object>().ClearInnerArray();
    }
}
```

Really, $etypeof(array) in proxy works incorrectly with inner generic classes - it use open type, instead of right closed generic type. Here is JS translation:

```
  var $T01 = function () {
    return ($T01 = JSIL.Memoize($asm00.GenericType$b1_InnerType)) ();
  };

  function GenericType$b1__ctor () {
    this.innerArray = JSIL.Array.New($asm00.GenericType$b1_InnerType.Of($thisType.T.get(this)), [new ($asm00.GenericType$b1_InnerType.Of($thisType.T.get(this))) ()]);
  };

  function GenericType$b1_ClearInnerArray () {
    JSIL.Array.Erase(this.innerArray, $T01().__Type__, 0, 1);
  };
```

In this example, correct type resolution was used for array creation and incorrect for Array.Clear.

**Comment by iskiselev** [comment]
I've fixed it - will do PR later today.

**Comment by iskiselev** [comment]
Fixed via #786.
